### ISSUE TITLE
Sass 3.3.2 --watch recompiles all .scss files, even if only one file changed

### ISSUE THREAD

**Issue by dutchwebworks** [sass/sass#1164]
The previous Sass 3.2.12 `--watch` command only recompiles changed files: `style.scss` or say `_header.scss` etc. partial files.

I've been playing with Sass 3.3.0, 3.3.1 and 3.3.2 on a big project last couple of days, which has lots of partials and normal `style.scss` like files. But the `--watch` command of Sass 3.3.2 recompiles everyting when only one files changed.

**Comment by nex3** [comment]
I can't reproduce this. Can you provide more details of your setup?

Here's what I did:

```
$ mkdir tmp out
$ echo 'partial1 {a: b}' > tmp/_partial1.scss
$ echo '@import "partial1"; test1 {a: b}' > tmp/test1.scss
$ echo 'partial2 {a: b}' > tmp/_partial2.scss
$ echo '@import "partial2"; test2 {a: b}' > tmp/test2.scss
$ sass --watch tmp:out &
>>> Sass is watching for changes. Press Ctrl-C to stop.
      write out/test1.css
      write out/test2.css
$ echo 'partial1 {a: c}' > tmp/_partial1.scss
>>> Change detected to: tmp/_partial1.scss
      write out/test1.css
$ echo '@import "partial2"; test2 {a: c}' > tmp/test2.scss
>>> Change detected to: tmp/test2.scss
      write out/test2.css
```

As expected, only the output files affected by the changes were recompiled. If you run the steps above, do you still see the problem?

**Comment by dutchwebworks** [comment]
Hi Nathan,

I've upgraded to Sass 3.3.3 on both Mac (10.9.2) and Windows (8.1). I've tested your demo project (as described in your post above). On the Mac I can't reproduce this bug anymore. Even on my large project as described above: in both projects **only the changed Sass (or parials) are recompiled** when using `--watch`. That's good!

I've copied over your demo project to my **Windows 8.1** machine, and ran the same command line as on Mac, but when using `--watch` the Windows machines **keeps recompiling all Sass files**, even if only one changed. The same goes for the large scale project as described in my initial post above (Windows keeps recompiling it every time). 

Below is some command-line output from Windows running Sass 3.3.3.

```
C:\Users\dennis\Downloads\sass-test-nathan>sass --watch .:out --style expanded --sourcemap
>>> Sass is watching for changes. Press Ctrl-C to stop.
      write out/test1.css
      write out/test1.css.map
      write out/test2.css
      write out/test2.css.map
      write out/test3.css
      write out/test3.css.map
>>> Change detected to: _partial1.scss
      write out/test1.css
      write out/test1.css.map
      write out/test2.css
      write out/test2.css.map
      write out/test3.css
      write out/test3.css.map
>>> Change detected to: test1.scss
      write out/test1.css
      write out/test1.css.map
      write out/test2.css
      write out/test2.css.map
      write out/test3.css
      write out/test3.css.map
```

And some info of my Windows machine

```
C:\Users\dennis>gem list -l

*** LOCAL GEMS ***

bigdecimal (1.2.0)
celluloid (0.15.2)
celluloid-io (0.15.0)
chunky_png (1.2.9)
coffee-script (2.2.0)
coffee-script-source (1.6.3)
compass (0.12.2)
execjs (2.0.2)
ffi (1.9.3 x64-mingw32)
fssm (0.2.10)
haml (4.0.4)
io-console (0.4.2)
json (1.7.7)
listen (2.7.0, 2.4.1)
minitest (4.3.2)
nio4r (1.0.0)
psych (2.0.0)
rake (0.9.6)
rb-fsevent (0.9.4)
rb-inotify (0.9.3)
rdoc (4.0.0)
sass (3.3.3, 3.2.12)
susy (1.0.9)
test-unit (2.0.0.0)
tilt (1.4.1)
timers (1.1.0)

C:\Users\dennis>ruby -v
ruby 2.0.0p247 (2013-06-27) [x64-mingw32]
```

The previous version we'd use was Sass 3.2.12 (both Mac & Windows) that didn't recompile everything 
when only one file was changed.

I did a `gem list -l` on Mac, and both Mac and Windows use `listen` v2.7.0.

**Comment by panec** [comment]
I have the same issue on Windows 7 with following gems:

```
$ gem list -l

*** LOCAL GEMS ***

bigdecimal (1.2.0)
celluloid (0.15.2)
chunky_png (1.2.8)
compass-import-once (1.0.2)
debase (0.0.9)
debugger-ruby_core_source (1.3.2)
ffi (1.9.3 x64-mingw32)
fssm (0.2.10)
gem (0.0.1.alpha)
io-console (0.4.2)
json (1.7.7)
listen (2.4.1, 1.1.6)
localtunnel (0.3)
minitest (4.3.2)
multi_json (1.8.4)
net-ssh (2.7.0)
net-ssh-gateway (1.2.0)
psych (2.0.0)
rake (0.9.6)
rb-fsevent (0.9.4)
rb-inotify (0.9.3)
rb-kqueue (0.2.0)
rdoc (4.0.0)
ruby-debug-ide (0.4.22)
rubygems-update (2.0.3)
sass (3.3.3)
test-unit (2.0.0.0)
timers (1.1.0)
wdm (0.1.0)

$ ruby -v
ruby 2.0.0p353 (2013-11-22) [x64-mingw32]
```

**Comment by nex3** [comment]
I've tried reproducing this on Windows, also unsuccessfully. It's possible that it's specific to listen versions other than the one bundled with Sass... can you try running `gem uninstall --all listen` and see if it starts working?

**Comment by nex3** [comment]
Never mind, I've managed to reproduce and track this down.

**Comment by nex3** [comment]
Fixed by 9b10d438bd0ba7c54453e266b6459d195a217e34.

**Comment by dutchwebworks** [comment]
Thank you very much for the fix! 

I've tested `Sass 3.3.4` in the demo project above and on my large scale project on both Mac and Windows. Bug is fixed, **Windows no longer recompiles** every *.scss file when only one changed.
